cadence/contracts/DapperStorageRent.cdc

```diff
@@ -0,0 +1,247 @@
+
+  /// getStorageRentRefillThreshold
+  /// Get the current StorageRentRefillThreshold
+  ///
+  /// @return UInt64 value of the current StorageRentRefillThreshold value
+  pub fun getStorageRentRefillThreshold(): UInt64 {
+    return self.StorageRentRefillThreshold
+  }
+
+  /// getRefilledAccounts
+  /// Get the current StorageRentRefillThreshold
+  ///
+  /// @return List of refilled Accounts
+  pub fun getRefilledAccounts(): [Address] {
+    return self.RefilledAccounts
+  }
+
+  /// getBlockedAccounts
+  /// Get the current StorageRentRefillThreshold
+  ///
+  /// @return List of blocked accounts
+  pub fun getBlockedAccounts() : [Address] {
+    return self.BlockedAccounts
+  }
+
+  /// getRefilledAccountInfos
+  /// Get the current StorageRentRefillThreshold
+  ///
+  /// @return Address: RefilledAccountInfo mapping
+  pub fun getRefilledAccountInfos(): {Address: RefilledAccountInfo} {
+    return self.RefilledAccountInfos
+  }
+
+  /// getRefillRequiredBlocks
+  /// Get the current StorageRentRefillThreshold
+  ///
+  /// @return UInt64 value of the current RefillRequiredBlocks value
+  pub fun getRefillRequiredBlocks(): UInt64 {
+    return self.RefillRequiredBlocks
+  }
+
+  /// tryRefill
+  /// Attempt to refill an accounts storage capacity if it has dipped below threshold and passes other checks.
+  ///
+  /// @param address: Address to attempt a storage refill on
+  pub fun tryRefill(_ address: Address) {
+    let REFUEL_AMOUNT = 0.06;
+
+    self.cleanExpiredRefilledAccounts(10)
+
+    // Get the Flow Token reciever of the address
+    let recipient = getAccount(address)
+    let receiverRef = recipient.getCapability<&PrivateReceiverForwarder.Forwarder>(PrivateReceiverForwarder.PrivateReceiverPublicPath).borrow()
+
+    // Sliently fail if the receiverRef is nill
+    if receiverRef == nil || receiverRef!.owner == nil {
+      emit RefilledFailed(address: address, reason: "Couldn't borrow the Accounts flowTokenVault")
+      return
+    }
+
+    // Sliently fail if the account has already be refueled within the block allowance
+    if self.RefilledAccountInfos[address] != nil && getCurrentBlock().height - self.RefilledAccountInfos[address]!.atBlock < self.RefillRequiredBlocks {
+      emit RefilledFailed(address: address, reason: "RefillRequiredBlocks")
+      return
+    }
+
+    // Get the users storage capacity and usage values
+    var low: UInt64 = recipient.storageUsed
+    var high: UInt64 = recipient.storageCapacity
+
+    if high < low {
+      high <-> low
+    }
+
+    // Sliently fail if the account has been blocked from receiving refills
+    if DapperStorageRent.getBlockedAccounts().contains(address) {
+      emit RefilledFailed(address: address, reason: "Address is Blocked")
+      return
+    }
+
+    // If the user is below the threshold PrivateReceiverForwarder will send 0.06 Flow tokens for about 6MB of storage
+    if high - low < self.StorageRentRefillThreshold {
```

Comment:
Remember that the value of `self.StorageRentRefillThreshold`  should be greater than the txn cost of the `tryRefil` transaction.